README.md

```diff
@@ -20,3 +20,39 @@ quantitatively reason with them.
 Add your answer to this markdown file. [This
 page](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions)
 might help with the notation for mathematical expressions.
+
+## Answer:
+
+**Defining a Good Pivot**: A good pivot in quicksort is one that roughly divides the array into two equal parts, which would ideally be the median or something close to it. A good pivot maximizes efficiency by minimizing the depth of the recursive calls.
+
+## First-Element Method:
+When we choose the first element:
+
+- The probability of it being the median (ideal pivot) is $1/n$ since any of the $n$ elements can be the median.
+- The probability of it being a good pivot (defined as being within the middle 50% of the sorted elements) is $1/2$. This is because there's an equal chance of it falling into the lower half or the upper half of the array.
+
+## Median-of-Three Method:
+When using the median-of-three method:
+
+- We select the first, middle and last elements of the array segment being sorted.
+- Each of these elements has an independent $1/2$ probability of being a good pivot.
+- The median of these three will be considered a good pivot unless all three are in the lower 25% or the upper 25% of the array. 
+  
+
+### Probability Analysis for Median-of-Three:
+Calculating the probability that the median-of-three is a good pivot choice:
+- **All Three Elements Are Good Pivots**: The probability is $\left(\frac{1}{2}\right)^3 = \frac{1}{8}$, because each has a $\frac{1}{2}$ chance of being in the desirable middle 50%.
+- **Two Elements Are Good Pivots, One is Bad**: There are three combinations for this, each with a probability of $\left(\frac{1}{2}\right)^2 * \left(\frac{1}{2}\right) = \frac{1}{8}$  With three possible combinations, the total probability is $3 \times \frac{1}{8} = \frac{3}{8}$.
+- **One Element is a Good Pivot, Two Are Bad**: The median can still be a good pivot if the single good element is the median among the three. There are three combinations for this, each with a probability of $\left(\frac{1}{2}\right) * \left(\frac{1}{2}\right)^2 = \frac{1}{8}$. However, since the good pivot must be the median, we multiply by $\frac{1}{3}$, giving us $3 * \frac{1}{8} * \frac{1}{3} = \frac{1}{8}$
```

Comment:
Not following this step. You multiply $3$ and $\frac{1}{3}$, giving you $1$ and the same probability as before?